### PR TITLE
[ios] set minimum ios version to 11.0, fix deprecations and cleanup dependencies

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Set minimum supported iOS version of 11.0. Removes support for 32bit iOS devices (which covers iPad 1-4 and iPhone 5 and earlier).

## Motivation and Context

Simplifies patching a logic around the iOS platform, especially around addons.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [X] **None of the above** (please explain below)

Not quite sure how to categorise this.

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
Early days for this one. Plenty more to do.

**Comment by kambala-decapitator** [comment]
- tools/depends/target/fontconfig/fix-aarch64_atomics.patch:17 (if this is Kodi's patch)
- xbmc/platform/darwin/ios/XBMCController.mm:710

**Comment by phunkyfish** [comment]
@kambala-decapitator the first is still required for 64bit if I'm not mistaken. For the second itis already set to 11.0 which is what it should be, correct?

**Comment by kambala-decapitator** [comment]
oh right, first is patch :)

in the second you should remove the condition, as it's always true now.

**Comment by phunkyfish** [comment]
I assume I can condense it to this:
`return UIEdgeInsetsInsetRect(self.view.bounds, m_window.safeAreaInsets);`

**Comment by phunkyfish** [comment]
@Rechi should removing ios from Jenkins be part of the PR or would that be something that must be done manually on/before merge?

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/a353ae2276a647f4f30f7fcf86baac6afc124d6c/tools/depends/configure.ac#L360-L372

Can be rewritten to something like 

```
        use_cpu=arm64
        platform_min_version="$target_platform-version-min=11.0"
        if test "$use_platform" = "tvos"; then
          target_platform=appletvos
        else
          target_platform=iphoneos
        fi
```

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/a353ae2276a647f4f30f7fcf86baac6afc124d6c/tools/depends/configure.ac#L378-L396

Can be rewritten like 

```
        case $use_sdk in
          11.*);;
          12.*);;
          13.*);;
            *)
              AC_MSG_ERROR(error in configure of --with-sdk=$use_sdk)
            ;;
        esac

```

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/a353ae2276a647f4f30f7fcf86baac6afc124d6c/tools/depends/configure.ac#L400-L403

can be removed

**Comment by fuzzard** [comment]
There is a line in README.iOS.md under section 2. Prequisites

```
Device with iOS 9.0 or newer to install Kodi after build.
```

Will need updating.

**Comment by fuzzard** [comment]
End of section 4 in the same readme, theres a NOTE that references an ios 9 build.

```
NOTE: Advanced developers may want to specify an iOS SDK version (if multiple versions are installed) in the configure line(s) shown above. The example below would use the iOS SDK 9.0:

./configure --host=arm-apple-darwin --with-cpu=arm64 --with-sdk=9.0
```

Probably need to just alter that to 11

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/xbmc/platform/darwin/DarwinUtils.mm#L178-L179

should be removable. The line below it should most likely stay as i believe its relevant for ios10/11/12

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/buildsteps/ios/package#L10-L15

no need for the check now with a single ios platform

```
UPLOAD_FILENAME="kodi-$(getBuildRevDateStr)-ios64.deb"
```

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in#L31-L34

No need for the check, just set ARM64

**Comment by fuzzard** [comment]
Can probably remove the previous ARM64 check completely, and then

https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in#L69-L70

Remove the $ARM64 && as every package will be 64bit

**Comment by fuzzard** [comment]
https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in#L80-L89

Change to just

```
echo "Package: $PACKAGE_ARM64"                  >  $DIRNAME/$PACKAGE/DEBIAN/control
echo "Name: @APP_NAME@-${PP_DEVICE} (64-bit)"   >> $DIRNAME/$PACKAGE/DEBIAN/control
echo "Pre-Depends: cy+cpu.arm64"                >> $DIRNAME/$PACKAGE/DEBIAN/control
echo "Conflicts: $PACKAGE"                      >> $DIRNAME/$PACKAGE/DEBIAN/control
echo "Replaces: $PACKAGE"                       >> $DIRNAME/$PACKAGE/DEBIAN/control
```

**Comment by fuzzard** [comment]
Remove

https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/darwin/Support/Codesign.command#L10

This can be reduced to just the LDID64 lines

https://github.com/xbmc/xbmc/blob/d14495a880c14a1c703a26ab96120108d514feee/tools/darwin/Support/Codesign.command#L14-L19

**Comment by Rechi** [comment]
First get this PR ready.
Once it is approved adjust the Jenkins configuration, run another PR build and if that reports success merge the PR.

File: docs/README.iOS.md [hunk 1]
```diff
@@ -75,25 +76,17 @@ git clone https://github.com/xbmc/xbmc kodi
 **[back to top](#table-of-contents)**
 
 ## 4. Configure and build tools and dependencies
-Kodi can be built as either a 32bit or 64bit program. The dependencies are built in `$HOME/kodi/tools/depends` and installed into `/Users/Shared/xbmc-depends`.
+Kodi can be built as a 64bit program for iOS. The dependencies are built in `$HOME/kodi/tools/depends` and installed into `/Users/Shared/xbmc-depends`.
 
 **TIP:** Look for comments starting with `Or ...` and only execute the command(s) you need.
 
 Configure build for 64bit (**recommended**):
```
Comment (Rechi) [inline docs/README.iOS.md]: What is the not recommended alternative?

File: tools/buildsteps/defaultenv [hunk 2]
```diff
@@ -22,7 +22,7 @@ case $XBMC_PLATFORM_DIR in
     DEFAULT_SDK_VERSION=12.2
     DEFAULT_XBMC_DEPENDS_ROOT=$WORKSPACE/tools/depends/xbmc-depends
     DEFAULT_CONFIGURATION="Debug"
-    DEFAULT_DARWIN_ARM_CPU="armv7"
+    DEFAULT_DARWIN_ARM_CPU="arm64"
```
Comment (Rechi) [inline tools/buildsteps/defaultenv]: Is there any case where the darwin arm cpu isn't arm64?

File: tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in [hunk 3]
```diff
@@ -9,7 +9,7 @@ SWITCH=`echo $1 | tr [A-Z] [a-z]`
 DIRNAME=`dirname $0`
 DSYM_TARGET_DIR=/Users/Shared/xbmc-depends/dSyms
 DSYM_FILENAME=@APP_NAME@.app.dSYM
-ARM64=false
+ARM64=true
```
Comment (Rechi) [inline tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in]: Is there any case where ARM64 isn't true?

File: tools/depends/configure.ac [hunk 4]
```diff
@@ -357,52 +357,29 @@ case $host in
         MC_CHECK_NOT_CPU([$use_cpu], "*86")
 
         # setup which sdk to use
+        use_cpu=arm64
```
Comment (Rechi) [inline tools/depends/configure.ac]: Keep the possibility to override the cpu.

File: docs/README.iOS.md [hunk 5]
```diff
@@ -103,9 +96,9 @@ make -j$(getconf _NPROCESSORS_ONLN)
 
 **WARNING:** Look for the `Dependencies built successfully.` success message. If in doubt run a single threaded `make` command until the message appears. If the single make fails, clean the specific library by issuing `make -C target/<name_of_failed_lib> distclean` and run `make`again.
 
-**NOTE:** **Advanced developers** may want to specify an iOS SDK version (if multiple versions are installed) in the configure line(s) shown above. The example below would use the iOS SDK 9.0:
+**NOTE:** **Advanced developers** may want to specify an iOS SDK version (if multiple versions are installed) in the configure line(s) shown above. The example below would use the iOS SDK 11.0:
 ```
-./configure --host=arm-apple-darwin --with-cpu=arm64 --with-sdk=9.0
+./configure --host=arm-apple-darwin --with-cpu=arm64 --with-sdk=11.0
```
Comment (Rechi) [inline docs/README.iOS.md]: Why pass `--with-cpu=arm64`?

File: tools/depends/target/ffmpeg/CMakeLists.txt [hunk 6]
```diff
@@ -64,9 +64,7 @@ elseif(CORE_SYSTEM_NAME STREQUAL android)
   endif()
   list(APPEND ffmpeg_conf --target-os=linux --extra-libs=-liconv --disable-linux-perf)
 elseif(CORE_SYSTEM_NAME STREQUAL darwin_embedded)
-  if(NOT CPU MATCHES arm64)
-    list(APPEND ffmpeg_conf --cpu=cortex-a8)
-  else()
+  if(CPU MATCHES arm64)
```
Comment (Rechi) [inline tools/depends/target/ffmpeg/CMakeLists.txt]: Is there any case where CPU isn't arm64?

File: tools/depends/target/ffmpeg/Makefile [hunk 7]
```diff
@@ -44,10 +44,7 @@ ifeq ($(OS), android)
   ffmpg_config += --target-os=linux --extra-libs=-liconv --disable-linux-perf
 endif
 ifeq ($(OS), darwin_embedded)
-  ifneq ($(CPU), arm64)
-    ffmpg_config += --cpu=cortex-a8
-    ffmpg_config += --as="$(NATIVEPREFIX)/bin/gas-preprocessor.pl -- $(CC)"
-  else
+  ifeq ($(CPU), arm64)
```
Comment (Rechi) [inline tools/depends/target/ffmpeg/Makefile]: Is there any case where CPU isn't arm64?

**Review by Rechi — CHANGES_REQUESTED** [review]
- ldid for 32 bit is still downloaded (extracted and installed)
- deployment target isn't updated
- There is still a dependency which explicitly disables a function, which is now available.

File: docs/README.iOS.md (hunk 1 above)
Comment (phunkyfish) [reply docs/README.iOS.md]: Fair point. Removed the recommended.

File: tools/depends/target/ffmpeg/CMakeLists.txt (hunk 6 above)
Comment (phunkyfish) [reply tools/depends/target/ffmpeg/CMakeLists.txt]: I don't believe so, but if there is then I should revert this?

File: tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in (hunk 3 above)
Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in]: I don't believe so

File: tools/depends/native/ldid/Makefile [hunk 8]
```diff
@@ -7,23 +7,16 @@ DEPS= ../../Makefile.include Makefile
 LIBNAME=ldid
 VERSION32=1.0.0
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: unused?

File: tools/depends/native/ldid/Makefile [hunk 9]
```diff
@@ -7,23 +7,16 @@ DEPS= ../../Makefile.include Makefile
...
 VERSION64=2
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: Why 64?

File: tools/depends/native/ldid/Makefile [hunk 10]
```diff
@@ -7,23 +7,16 @@ DEPS= ../../Makefile.include Makefile
...
-SOURCE32=$(LIBNAME)-$(VERSION32)
 SOURCE64=$(LIBNAME)-$(VERSION64)
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: Why 64?

File: tools/depends/native/ldid/Makefile [hunk 11]
```diff
@@ -7,23 +7,16 @@ DEPS= ../../Makefile.include Makefile
...
-ARCHIVE32=$(SOURCE32).tar.gz
 ARCHIVE64=$(SOURCE64).tar.gz
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: Why 64?

File: tools/depends/native/ldid/Makefile [hunk 12]
```diff
@@ -7,23 +7,16 @@ DEPS= ../../Makefile.include Makefile
...
 $(TARBALLS_LOCATION)/$(ARCHIVE64):
 	cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) $(RETRIEVE_TOOL_FLAGS) $(BASE_URL)/$(ARCHIVE64)
 
-$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE32) $(TARBALLS_LOCATION)/$(ARCHIVE64) $(DEPS)
+$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE64) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
-	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE32)
-	mv $(PLATFORM)/ldid $(PLATFORM)/ldid32
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE64)
 	mv $(PLATFORM)/ldid $(PLATFORM)/ldid64
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: Why 64?

Comment (phunkyfish) [reply tools/depends/native/ldid/Makefile]: Fixed, also updated command call in codesign command.

File: tools/depends/native/ldid/Makefile [hunk 13]
```diff
@@ -5,27 +5,19 @@ DEPS= ../../Makefile.include Makefile
...
-
-$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE32) $(TARBALLS_LOCATION)/$(ARCHIVE64) $(DEPS)
+$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
-	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE32)
-	mv $(PLATFORM)/ldid $(PLATFORM)/ldid32
-	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE64)
-	mv $(PLATFORM)/ldid $(PLATFORM)/ldid64
+	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+	mv $(PLATFORM)/ldid $(PLATFORM)/ldid
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: Why rename to the same name?

File: tools/depends/native/ldid/Makefile [hunk 14]
```diff
@@ -5,27 +5,19 @@ DEPS= ../../Makefile.include Makefile
...
 
 .installed-$(PLATFORM):$(PLATFORM)
 	cp $(PLATFORM)/ldid* $(NATIVEPREFIX)/bin
```
Comment (Rechi) [inline tools/depends/native/ldid/Makefile]: `ldid*`?

File: tools/depends/native/ldid/Makefile (hunk 13 above)
Comment (phunkyfish) [reply tools/depends/native/ldid/Makefile]: Done

File: tools/depends/native/ldid/Makefile (hunk 14 above)
Comment (phunkyfish) [reply tools/depends/native/ldid/Makefile]: Done

**Comment by phunkyfish** [comment]
@Rechi should be good now.

File: xbmc/platform/darwin/ios/Info.plist.in [hunk 15]
```diff
@@ -119,7 +119,7 @@
 	<true/>
 	<key>UIRequiredDeviceCapabilities</key>
 	<array>
-		<string>armv7</string>
+		<string>armv64</string>
```
Comment (Rechi) [inline xbmc/platform/darwin/ios/Info.plist.in]: Are you sure?

**Review by Rechi — CHANGES_REQUESTED** [review]
- Jenkins depends configuration differs from what the readme tells.
- generated DEBIAN/control file looks wrong
- There is a patch hunk which worked around a function, which is now available.
- 1 new warning
- host is now actually `aarch64-apple-darwin` and not `arm-apple-darwin`

File: xbmc/platform/darwin/ios/Info.plist.in (hunk 15 above)
Comment (phunkyfish) [reply xbmc/platform/darwin/ios/Info.plist.in]: No I’m not

Comment (phunkyfish) [reply xbmc/platform/darwin/ios/Info.plist.in]: I am now: https://developer.apple.com/documentation/bundleresources/information_property_list/uirequireddevicecapabilities

Comment (phunkyfish) [reply xbmc/platform/darwin/ios/Info.plist.in]: Whoops, extra `v`

**Comment by phunkyfish** [comment]
> * Jenkins depends configuration differs from what the readme tells.
> * generated DEBIAN/control file looks wrong
> * There is a patch hunk which worked around a function, which is now available.
> * 1 new warning
> * host is now actually `aarch64-apple-darwin` and not `arm-apple-darwin`

Ok, I fixed the warning and renamed armv64 to arm64 in the plist, also set the default darwin arm cpu correctly.

I'll need your help with these as I'm unsure what they should be:

* generated DEBIAN/control file looks wrong. Please help here.
* There is a patch hunk which worked around a function, which is now available. I've searching but i can't find this.
* host is now actually `aarch64-apple-darwin` and not `arm-apple-darwin`. Is arm-apple-darwin what it should be?

File: tools/buildsteps/defaultenv [hunk 16]
```diff
@@ -4,7 +4,7 @@ NDK_VERSION=${NDK_VERSION:-"Default"}
 NDK_API=${NDK_API:-"Default"}
 Configuration=${Configuration:-"Default"}
 XBMC_DEPENDS_ROOT=${XBMC_DEPENDS_ROOT:-"Default"}
-DARWIN_ARM_CPU=${DARWIN_ARM_CPU:-"Default"}
+DARWIN_ARM_CPU=${DARWIN_ARM_CPU:-"arm64"}
```
Comment (Rechi) [inline tools/buildsteps/defaultenv]: Is there any case where the darwin arm cpu isn't arm64?

**Comment by Rechi** [comment]
- DEBIAN/control
```sh
echo "Package: $PACKAGE"                  >  $DIRNAME/$PACKAGE/DEBIAN/control
# [...]
echo "Conflicts: $PACKAGE"                      >> $DIRNAME/$PACKAGE/DEBIAN/control
echo "Replaces: $PACKAGE"                       >> $DIRNAME/$PACKAGE/DEBIAN/control
```
- `git grep -i clock_gettime -- tools/depends/`
- Currently one configures with `--host=arm-apple-darwin`, but the result are actually binaries for `aarch64-apple-darwin`.

File: tools/buildsteps/defaultenv (hunk 16 above)
Comment (fuzzard) [reply tools/buildsteps/defaultenv]: Not currently

Comment (kambala-decapitator) [reply tools/buildsteps/defaultenv]: there exists `arm64e`, but it can run `arm64` code

Comment (phunkyfish) [reply tools/buildsteps/defaultenv]: I understand something should be changed because it's the only option I just don't know what you would like changed.

**Comment by phunkyfish** [comment]
Thanks for patch hint. Found it now.

**Comment by phunkyfish** [comment]
Ok, just these left:

- tools/buildsteps/defaultenv - arm64
- Currently one configures with `--host=arm-apple-darwin`, but the result are actually binaries for `aarch64-apple-darwin`.

File: tools/buildsteps/defaultenv (hunk 16 above)
Comment (Rechi) [reply tools/buildsteps/defaultenv]: Replace the usage of `DARWIN_ARM_CPU` with it's actual value.

**Comment by Rechi** [comment]
The resulting binaries are correct, they are the same as before. But arm CPU is arm for 32 bit and aarch64 CPU is arm for 64 bit.

**Comment by phunkyfish** [comment]
So do you mean we just need to  update the  docs so the host is specified as this?

Would this not be the same for tvOS?

File: tools/buildsteps/ios/configure-depends [hunk 17]
```diff
@@ -8,6 +8,6 @@ then
     --with-tarballs=/Users/Shared/xbmc-depends/tarballs \
     --host=arm-apple-darwin \
     --with-sdk=$SDK_VERSION \
-    --with-cpu=$DARWIN_ARM_CPU \
+    --with-cpu=arm64 \
```
Comment (Rechi) [inline tools/buildsteps/ios/configure-depends]: Why does this configure in a different way then the documentation tells?

**Comment by Rechi** [comment]
Updating the documentation is only a part of it. Yes it's the same for tvOS.

**Comment by phunkyfish** [comment]
So cpu should be set to aarch64 across the board for darwin (iOS/tvOS)? arm64 is just incorrect.

**Comment by Rechi** [comment]
The host triple `arm-apple-darwin`, was always wrong for arm64/aarch64 builds. As now there are only those kind of builds, the host triple should be reflect that.
You'll also find hacks, because of the wrong host triple usage.

**Comment by phunkyfish** [comment]
Ok, so if I do that as part of this PR it will mean also making tvOS changes. That’s ok to do?

**Comment by phunkyfish** [comment]
So a question. Where exactly should it be set to arm64 and where to aarch64?

Or is just aarch64 everywhere instead of arm/arm64?

**Comment by Rechi** [comment]
You can do the changes in this PR. Preferred is to have a commit fixing the host triple name before the minimum version bump, but you can combine it into the minimum version bump commit.
The CPU part of the host triplet has to be aarch64. Everywhere else you can keep arm64.

**Comment by phunkyfish** [comment]
@Rechi - I may need your help sourcing the whereabouts of the hacks. I only found one on the host triplet but clearly there are many more.

**Comment by Rechi** [comment]
I only found one more, `git grep aarch64-apple-darwin`.

**Comment by phunkyfish** [comment]
Hmmm, strange. That's the command I used but I didn't find another one with it.

**Comment by phunkyfish** [comment]
Anyone have any idea? After changing the host triplet in the second commit both ios64 and tvOS both fail on gnutls.

```
09:57:53 make[8]: *** No rule to make target `macosx/sha1-armv8.s', needed by `macosx/sha1-armv8.lo'.  Stop.
09:57:53 make[8]: *** Waiting for unfinished jobs....
```

**Comment by fuzzard** [comment]
check slack for a diff to work around the fault.

File: tools/depends/target/gnutls/Makefile [hunk 18]
```diff
@@ -28,9 +28,9 @@ $(TARBALLS_LOCATION)/$(ARCHIVE):
 
 $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
-	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
 	cd $(PLATFORM); patch -p1 -i ../size-max.patch
 	cd $(PLATFORM); patch -p1 -i ../add-dl-as-private-lib.patch
+	cd $(PLATFORM); patch -p1 -i ../disable-hw-accel-aarch64.patch
```
Comment (Rechi) [inline tools/depends/target/gnutls/Makefile]: Why disable hardware acceleration for all aarch64 cpus?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Good point

File: tools/depends/target/config.site.in [hunk 19]
```diff
@@ -106,6 +95,10 @@ fi
 # Python3 build for appletvos
 if test "@target_platform@" = "appletvos"; then
   ac_cv_func_sigaltstack=no
+  if test "${PACKAGE_NAME}" = "gnutls"; then
+    ac_cv_func_vfork_works=no
+    ac_cv_func_fork=no
```
Comment (Rechi) [inline tools/depends/target/config.site.in]: Does `fork` and `vfork` work in general on appletvos?

Comment (phunkyfish) [reply tools/depends/target/config.site.in]: No, it appears not.

Comment (Rechi) [reply tools/depends/target/config.site.in]: In that case, why is it only disabled for a package named gnutls?

File: tools/depends/target/gnutls/Makefile [hunk 20]
```diff
@@ -1,23 +1,16 @@
...
 
 # lib name, version
 LIBNAME=gnutls
 VERSION=3.6.11.1
 SOURCE=$(LIBNAME)-$(VERSION)
 ARCHIVE=$(SOURCE).tar.xz
 
-ifeq (darwin, $(findstring darwin, $(HOST)))
-# darwins tar doesn't know about xz - so we need our native version of tar here
-ARCHIVE_TOOL=$(ARCHIVE_TOOL_NATIVE)
```
Comment (Rechi) [inline tools/depends/target/gnutls/Makefile]: How is this change related to fixing the host triplet to `aarch64-apple-darwin`?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Moving to separate commit

Comment (Rechi) [reply tools/depends/target/gnutls/Makefile]: Isn't this variable (`ARCHIVE_TOOL_NATIVE`) now unused?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: So remove tar altogether from native depends?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Or is it needed to other stuff?

Comment (Rechi) [reply tools/depends/target/gnutls/Makefile]: If it is unused or not required, yes.

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: tar needed for dpkg on ios/tvos so it stays I guess.

https://github.com/xbmc/xbmc/blob/1eff5afda088f8fc58b63f8a7313c2b71016b3e2/tools/depends/native/Makefile#L16-L18

Comment (Rechi) [reply tools/depends/target/gnutls/Makefile]: Is it really needed by dpkg, or did only previously system tar versions miss some features?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Let’s remove it and see

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Fails on darwin_embedded with it removed. Must be required.

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: Hmm, or not. I forgot to remove the dependency.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
jenkins build this please

File: tools/depends/target/gnutls/Makefile (hunk 20 above)
Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: tar/xz still needed for ios/tvos.

**Comment by phunkyfish** [comment]
@Rechi, this should be everything now i think

File: tools/depends/target/gnutls/Makefile [hunk 21]
```diff
@@ -31,6 +24,9 @@ $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
 	cd $(PLATFORM); patch -p1 -i ../size-max.patch
 	cd $(PLATFORM); patch -p1 -i ../add-dl-as-private-lib.patch
+ifeq ($(OS),darwin_embedded)
+	cd $(PLATFORM); patch -p1 -i ../disable-hw-accel-aarch64.patch
```
Comment (Rechi) [inline tools/depends/target/gnutls/Makefile]: Why patch the buildsystem, instead of usinging the already existing configuration switch?

Comment (phunkyfish) [reply tools/depends/target/gnutls/Makefile]: I didn't realise there was a config switch. Will change that tomorrow.

**Comment by phunkyfish** [comment]
Jenkins build this please

**Comment by phunkyfish** [comment]
Ok, @Rechi, think it's ready now for the jenkins changes now.

File: tools/depends/target/config.site.in [hunk 22]
```diff
@@ -103,9 +92,12 @@ if test "@platform_os@" = "darwin_embedded"; then
   fi
 fi
 
-# Python3 build for appletvos
 if test "@target_platform@" = "appletvos"; then
+  # Python3 build
   ac_cv_func_sigaltstack=no
+  # gnutls build
```
Comment (Rechi) [inline tools/depends/target/config.site.in]: Why add the lib info? This info might get outdated if libraries are updated.
What useful information does this comment provide?

File: tools/depends/target/config.site.in [hunk 23]
```diff
@@ -103,9 +92,12 @@ if test "@platform_os@" = "darwin_embedded"; then
...
+  ac_cv_func_vfork_works=no
+  ac_cv_func_fork=no
```
Comment (Rechi) [inline tools/depends/target/config.site.in]: Doesn't that apply for binary add-ons too?

File: tools/depends/target/config.site.in (hunk 22 above)
Comment (phunkyfish) [reply tools/depends/target/config.site.in]: Easy to remove.

File: tools/depends/target/config.site.in (hunk 23 above)
Comment (phunkyfish) [reply tools/depends/target/config.site.in]: You mean also set this in `config-binaddons.site.in`. It wasn't required before now.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
Ok, ready @Rechi

**Comment by phunkyfish** [comment]
@Rechi is it ok to make the Jenkins changes now?

**Comment by lrusak** [comment]
> There is still a (new) hack because of the wrong host triplet.

Could you be more specific? Do you mean about the mixed usage of `arm64` and `aarch64`?

**Comment by phunkyfish** [comment]
Maybe I’ve been looking at this too long but I’m not seeing it.

**Comment by Rechi** [comment]
> This branch is 3 commits ahead, 113 commits behind xbmc:master.

https://github.com/phunkyfish/xbmc/tree/min-ios-version

So there are changes from 113 commits which are currently not included in your branch.

**Comment by phunkyfish** [comment]
Ah, that would make sense why I’m not seeing it so. At least I didn’t add the new hack!

**Comment by phunkyfish** [comment]
Ok, I removed the todo's in the two toolchain files. Any other hacks you came across @Rechi ?

I didn't spot anything additional.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
@Rechi is it ok to make the jenkins changes now?

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Review by lrusak — APPROVED** [review]
